Use basic auth to authenticate with 3scale Porta

jmprusi opened the pull request:
By using basic auth, we can send either access_token or provider_key as credentials
to authenticate api calls with porta.

eguzki commented on client/types.go:
```diff
@@ -111,6 +111,7 @@ type Plan struct {
 	State              string   `xml:"state"`
 	ServiceID          string   `xml:"service_id"`
 	EndUserRequired    string   `xml:"end_user_required"`
+	ApprovalRequired   string   `xml:"approval_required"`
```
Changes in `types.go` seem to be part of another PR

jmprusi replied:
this is built on top of the latest

PhilipGough commented:
@eguzki You are saying at client creation time, pass the credentials as a field to the client instead of passing it through all the public interfaces?

I'm ok with that if you think it will be better. You guys are the main consumers of this at the moment so whatever best fits your needs.

unleashed commented:
Skimmed over this. I wonder why use Basic auth instead of using URL parameters? With the latter monitoring is simpler.

Re: passing of credentials, I'd not pass them around in each call and I'd not make them a static property of the client, but instead I'd set a pointer to them in the client so I'd be able to swap them out whenever needed.

jmprusi commented:
@unleashed this is basically to avoid having to specify the type of credentials as a param name ( "access token" or "provider_key"). By using basic auth, Porta API will use both indistinctly.

unleashed commented:
@jmprusi kinda hacky on their part - we still lose visibility in the logs though, I'd say it's convenient from the POV of this client, but for people monitoring that makes things harder.

That said, they added it, so you could always blame that bad decision and have them be sorry for ops people running their software... :)

eguzki commented:
to be more flexible, pass credentials on client constructor and then the client provides methods to update credentials. This way you do not have to create new client for different credentials.

jmprusi commented:
@PhilipGough @unleashed @eguzki  Ok... just added the credential field to the client and modified the constructor, now we need to instantiate the client with the credential, also I added a method to change the client credential (SetCredential).
